File: CS/DxSample/MyPivotGrid.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraPivotGrid.Data;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPivotGrid.ViewInfo;

namespace DxSample {
    public class MyPivotGrid : PivotGridControl {
        public MyPivotGrid() : base() { }
        protected override PivotGridViewInfoData CreateData() {
            return new MyPivotGridViewInfoData(this);
        }
    }

    public class MyPivotGridField : PivotGridField {
        public MyPivotGridField(string fieldName, PivotArea area) : base(fieldName, area) { }
        public MyPivotGridField() : base() { }
        public MyPivotGridField(PivotGridData data) : base(data) { }
        
        protected override bool OnAllowSerializationProperty(DevExpress.Utils.OptionsLayoutBase options, string propertyName, int id) {
            if(propertyName == "Options")
                return false;
            return base.OnAllowSerializationProperty(options, propertyName, id);
        }
    }

    public class MyPivotGridFieldCollection : PivotGridFieldCollection {
        public MyPivotGridFieldCollection(PivotGridData data) : base(data) { }
        protected override PivotGridField CreateFieldCore(string fieldName, PivotArea area) {
            return new MyPivotGridField(fieldName, area);
        }
    }

    public class MyPivotGridViewInfoData : PivotGridViewInfoData {
        public MyPivotGridViewInfoData(IViewInfoControl control) : base(control) { }
        public MyPivotGridViewInfoData() : this(null) { }

        protected override PivotGridFieldCollectionBase CreateFieldCollection() {
            return new MyPivotGridFieldCollection(this);
        }
    }
}

File: VB/DxSample/MyPivotGrid.vb
Imports DevExpress.XtraPivotGrid.Data
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPivotGrid.ViewInfo

Namespace DxSample

    Public Class MyPivotGrid
        Inherits PivotGridControl

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function CreateData() As PivotGridViewInfoData
            Return New MyPivotGridViewInfoData(Me)
        End Function
    End Class

    Public Class MyPivotGridField
        Inherits PivotGridField

        Public Sub New(ByVal fieldName As String, ByVal area As PivotArea)
            MyBase.New(fieldName, area)
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal data As PivotGridData)
            MyBase.New(data)
        End Sub

        Protected Overrides Function OnAllowSerializationProperty(ByVal options As DevExpress.Utils.OptionsLayoutBase, ByVal propertyName As String, ByVal id As Integer) As Boolean
            If Equals(propertyName, "Options") Then Return False
            Return MyBase.OnAllowSerializationProperty(options, propertyName, id)
        End Function
    End Class

    Public Class MyPivotGridFieldCollection
        Inherits PivotGridFieldCollection

        Public Sub New(ByVal data As PivotGridData)
            MyBase.New(data)
        End Sub

        Protected Overrides Function CreateFieldCore(ByVal fieldName As String, ByVal area As PivotArea) As PivotGridField
            Return New MyPivotGridField(fieldName, area)
        End Function
    End Class

    Public Class MyPivotGridViewInfoData
        Inherits PivotGridViewInfoData

        Public Sub New(ByVal control As IViewInfoControl)
            MyBase.New(control)
        End Sub

        Public Sub New()
            Me.New(Nothing)
        End Sub

        Protected Overrides Function CreateFieldCollection() As PivotGridFieldCollectionBase
            Return New MyPivotGridFieldCollection(Me)
        End Function
    End Class
End Namespace
